src/main/java/rx/internal/operators/OperatorRetryWithPredicate.java

```diff
@@ -109,7 +109,7 @@ public void onNext(T v) {
                         };
                         // register this Subscription (and unsubscribe previous if exists) 
                         serialSubscription.set(subscriber);
-                        o.unsafeSubscribe(subscriber);
+                        o.subscribe(subscriber);
```

Comment:
Instead of this, I'd add a done flag to the inner Subscriber to ignore any subsequent events; should be more lightweight.